torch/_decomp/decompositions.py

```diff
@@ -1301,3 +1301,19 @@ def log_sigmoid_forward(self: Tensor) -> Tuple[Tensor, Tensor]:
     else:
         buffer = z
     return min - torch.log1p(z), buffer
+
+# torch.ops.aten.norm only supports numeric p, does not support Frobenius norm or nuclear norm
+@register_decomposition([aten.norm.ScalarOpt_dtype, aten.norm.Scalar, aten.norm.ScalarOpt_dim_dtype,
+        aten.norm.ScalarOpt_dim, aten.norm.names_ScalarOpt_dim_dtype, aten.norm.names_ScalarOpt_dim])
+@reduction_complex_to_real
+def norm(self: Tensor, p: float, dim: List[int] = [], keepdim: bool = False):
+    self_ = self.abs()
+    if p==0:
+        result = (self_ != 0).sum(dim, keepdim = keepdim)
+    elif p==float('inf'):
+        result = self_.amax(dim, keepdim = keepdim)
+    elif p==-float('inf'):
+        result = self_.amin(dim, keepdim = keepdim)
+    else:
+        result = self_.pow(p).sum(dim, keepdim = keepdim).pow(1/p)
+    return result.to(dtype=self.dtype)
```

Comment:
this will turn `result` to complex dtype again for complex inputs?

Reply:
I suspect for perf reasons you still need a special case for p=2 instead of doing `pow(p)` and `pow(1/p)`

Reply:
I think the result will not be in complex dtype for complex inputs because of the @reduction_complex_to_real decorator.

Reply:
I think the decorator makes it so that this type conversion is unnecessary though.

Reply:
oh sorry nvm will be a complex type. Fixed now in the new commit.

Reply:
> I think the decorator makes it so that this type conversion is unnecessary though.

Do you mean line 1319 can be just `return result` ? It works (passes all -k norm tests). I made this change in the newest commit.